vm.go

```diff
@@ -380,6 +382,20 @@ func (vm *vm) SendMessage(hyper *api.Hyper) error {
 	return err
 }
 
+var waitForShimTimeout = 30 * time.Second
+
+// WaitFormShim will wait until a shim claiming the ioSession has registered
+// itself with the proxy. If the shim has already done so, WaitForSim will
+// return immediatlely.
+func (session *ioSession) WaitForShim() error {
+	select {
+	case <-session.shimConnected:
+	case <-time.After(waitForShimTimeout):
```

Comment:
This seems like a reasonable timeout, but since it is a relatively long time, I think it could be useful to display a message every second or so that shows the status. Something like:

```
Waiting for shim to register itself with token %s (timeout in %d seconds)
```

That way, it'll be clear from the log what the proxy is doing and that it hasn't just "hung".